ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/LabelClickSpikes.java

```diff
@@ -46,11 +40,10 @@ public class LabelClickSpikes extends
   /**
    * Composite transform that wraps {@code DetectClickSpikes} with keying by {@code context_id}.
    */
-  public static PTransform<PCollection<PubsubMessage>, PCollection<PubsubMessage>> perContextId(
+  public static PTransform<PCollection<SponsoredInteraction>, PCollection<SponsoredInteraction>> perContextId(
       Integer maxClicks, Duration windowDuration) {
     return PTransform.compose("DetectClickSpikesPerContextId", input -> input //
-        .apply(WithKeys.of((message) -> message.getAttribute(Attribute.CONTEXT_ID))) //
-        .setCoder(KvCoder.of(StringUtf8Coder.of(), PubsubMessageWithAttributesCoder.of()))
+        .apply(WithKeys.of((interaction) -> interaction.getContextId())) //
```

Comment:
```suggestion
        .apply(WithKeys.of((interaction) -> interaction.getContextId())) //
        .setCoder(KvCoder.of(StringUtf8Coder.of(), SponsoredInteraction.getCoder())) //
```

When attempting to deploy to Dataflow, I got a coder error here.